dynamic_programming/house_robber.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file
+ * @brief Implementation of [House Robber
+ * Problem](https://labuladong.gitbook.io/algo-en/i.-dynamic-programming/houserobber)
+ * algorithm
+ * @details
+ * Solution of House robber problem uses a dynamic programming concept that
+ * works in \f$O(n)\f$ time and works in \f$O(1)\f$ space.
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <climits>   /// for std::max
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+namespace dynamic_programming {
+/**
+ * @namespace house_robber
+ * @brief Functions for the [House
+ * Robber](https://labuladong.gitbook.io/algo-en/i.-dynamic-programming/houserobber)
+ * algorithm
+ */
+namespace house_robber {
+/**
+ * @brief The main function implements house robber algorithm using dynamic
+ * programming
+ * @param money array containing money in the ith house
+ * @param n size of array
+ * @returns maximum amount of money that can be robbed
+ */
+std::uint32_t houseRobber(const std::vector<int> &money, const int &n) {
```

Comment:
```suggestion
std::uint32_t houseRobber(const std::vector<uint32_t> &money, const uint32_t &n) {
```
I think @Panquesito7 is saying to you may use specific data type everywhere now only in the return type of function.  This is just example you can do it according to the need.

Reply:
Got it now, since I am using the variables inside houseRobber function are for storing money values that is surely non-negative so its better to use uint32_t. I have done the required changes. Pl check it.